examples/delete_disks.py

```diff
@@ -0,0 +1,81 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2022 Red Hat, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""
+Show how to delete many disks created by the 'add_disks.py' example quickly.
+"""
+
+import time
+import sys
+from contextlib import closing
+
+import ovirtsdk4 as sdk
+
+from helpers import common
+from helpers.common import progress
+
+parser = common.ArgumentParser(description="Delete disks by given description")
+
+parser.add_argument(
+    "--sd-name",
+    required=True,
+    help="Name of the storage domain.")
+
+parser.add_argument(
+    "--description",
+    default="Created by add_disks.py",
+    help="The description of the disks to be deleted.")
+
+args = parser.parse_args()
+
+progress("Connecting...")
+connection = common.create_connection(args)
+with closing(connection):
+    storage_domains_service = connection.system_service().storage_domains_service()
+    found_sd = storage_domains_service.list(search=f'name={args.sd_name}')
+    if not found_sd:
+        raise RuntimeError(f"Cannot find storage domain {args.sd_name}")
+
+    sd = found_sd[0]
+    sd_service = storage_domains_service.storage_domain_service(sd.id)
+    sd_disks_service = sd_service.disks_service()
+    waiting = set()
+
+    # StorageDomainDisksService.list has no 'search' parameter and ignores
+    # query={'name': 'spam'} so we have to do the filtering ourselves
+    disks = [disk for disk in sd_disks_service.list()
+             if disk.description == args.description]
+
+    if not disks:
+        progress("No disks found")
+        sys.exit(1)
+
+    for disk in disks:
+        sd_disks_service.disk_service(disk.id).remove()
+        waiting.add(disk.id)
+        progress(f"Waiting for disk {len(waiting)} {disk.id} to be removed...")
```

Comment:
This log is confusing because we don't really wait for anything here, and the log does not match the log in the wait loop.

Maybe move the log before calling remove, and change it to:

    Removing disk {disk.id} ({len(waiting)}/{len(disks)})

And the waiting log to:

    Disk {disk_id} removed ({len(disks) - len(waiting)}/{len(disks)})

So the logs will be (assuming 3 disks):

    Removing disk aaa-bbb (1/3)
    Removing disk bbb-ccc (2/3)
    Removing disk ccc-ddd (3/3)
    Disk bbb-ccc removed (1/3)
    Disk aaa-bbb removed (2/3)
    Disk ccc-ddd removed (3/3)
   
Note that disks may be removed in different order.

Reply:
ack